test/system/test_tpm2_unseal.sh

```diff
@@ -105,7 +127,9 @@ tpm2_create -Q -g $alg_create_obj -G $alg_create_key -u $file_unseal_key_pub -r
 
 tpm2_load -Q -c $file_primary_key_ctx  -u $file_unseal_key_pub  -r $file_unseal_key_priv -n $file_unseal_key_name -C $file_unseal_key_ctx
 
-unsealed=`tpm2_unseal -c $file_unseal_key_ctx -L ${alg_pcr_policy}:${pcr_ids} -F $file_pcr_value`
+tpm2_unseal -c $file_unseal_key_ctx -L ${alg_pcr_policy}:${pcr_ids} -F $file_pcr_value > "$file_unsealed_data_out"
+
+unsealed=`yaml_get "sensitive" "$file_unsealed_data_out" | xxd -r`
```

Comment:
I like the yaml output for tools list tpm2_pcrlist or tpm2_nvlist but I think this is taking it too far.

It's quite likely that people will want tpm2_unseal to just return the same sensitive data that was provided to tpm2_create when the object was created. Having it print as a yaml output with a single `sensitive:` field just makes user's life more dificult IMHO.

Same for tpm2_read, it should just read whatever was written in the NVRAM with tpm2_write. From a user POV she just wrote some bytes and wants to read again the same.

Reply:
@martinezjavier I don't want raw bytes dumped to the terminal. Consider a user inspecting the contents and forgets to | to xxd or something, their going to end up with unescaped sequences hitting their terminal.

Reply:
@williamcroberts well, the user should know what's doing. It will be the same if tries to read any raw binary data.

The reason why I added tpm2_unseal to print the unsealed data to the stdout in the first place in commit 8aa7f644386d ("tpm2_unseal: allow to write unsealed data to the standard output") was to avoid the need to store sensitive data to disk.

But printing the data as a hexdump inside a yaml makes basically that information unusable. Since you now need to parse the data and transform, which means that you need tools like xxd and a yaml parser (or at least grep) to get what I think is very basic information (the data that you sealed).

I think that at the very least we need to support both, it's OK if you don't want to make printing the raw data as the default (although I would) but we need a command line option to allow users IMHO.

Reply:
I think my point is that tpm2_unseal, obtains the sensitive part of the sealed object and tpm2_nvread reads a NVRAM index area. The tool shouldn't try to infer what kind of data that is.

f that data are bytes with ASCII values, then it will be friendly to print in a terminal and if is binary data, then it won't. 

It's absolutely valid to seal (or nvwrite) ASCII bytes for example and that will be printed correctly on a terminal. So it would be weird for a user to get a hexdump of the string "foobar".

Just like any tools reading data and printing to stdout (i.e: cat) does. The user should know what kind of information is reading and redirecting the stdout to another file if doesn't want to end in the terminal.

Reply:
@martinezjavier let me explain further why I made this choice:
Consistent parsing scheme. As we  move forward, outputs will become backwards compatible and YAML makes this easy. We can add, arbitrarily to the outputs and the YAML code written to parse it will always work. I'm hoping the same for options, id like to get less churn in the releases as the code starts to become more robust, sane and well tested. Theirs still a long way to go, these tools were in quite awful state when I took over, and still are :-1: 

However, the tools you mention, it would be easy to add a -o <file> option that supports the "-" so you could send binary to stdout. This seems sane to me.

Reply:
@williamcroberts and one last argument against to output in yaml + hexdump is that it makes the operations non-reciprocal. As a user I would think why read doesn't work the same as write?

For example I wold be very confused with something like the following:

```
$ echo foobar > file.txt
$ cat file.txt
0000000 6f66 626f 7261 000a                    
0000007
```

As you can see from my lengthy responses, I've a very strong opinion on this :smiley:

Reply:
Now echo some non-ascii, this is where it falls apart at the shell.
```
$ echo -e "\x02\xc5\xd8" > foo
$ cat foo
�
```

Reply:
Yes, I know that. My point was that `cat` doesn't try to be smart and just prints to stdout whatever reads from the file. If the user wrote binary data there and tries to read it, it's their problem.

Reply:
I'm coming around to your argument. I'm gonna back out nvread to hexdump then as well as these yaml parts to these changes.

Ill change my stance to:
Any tool that has a compound output, is YAML. Other tools, data as is